docs/ldml/tr35.md

```diff
@@ -2191,18 +2191,22 @@ This operation is performed in the following way.
    5. Get the components of the cleaned-up source tag _(language<sub>s</sub>, script<sub>s</sub>,_ and _region<sub>s</sub>_), plus any variants and extensions.
 2. **Lookup.** Look up each of the following in order, and stop on the first match:
    1. _language<sub>s</sub>\_script<sub>s</sub>\_region<sub>s</sub>_
-   2. _language<sub>s</sub>\_region<sub>s</sub>_
-   3. _language<sub>s</sub>\_script<sub>s</sub>_
+   2. _language<sub>s</sub>\_script<sub>s</sub>_
+   3. _language<sub>s</sub>\_region<sub>s</sub>_
    4. _language<sub>s</sub>_
-   5. und\__script<sub>s</sub>_
 3. **Return**
-   1. If there is no match, either return
-      1.  an error value, or
-      2.  the match for "und" (in APIs where a valid language tag is required).
+   1. If there is no match, signal an error and stop.
```

Comment:
I think I find another bug in this algorithm. This algorithm does not return earlier if we already have LSR and do not need to maximize. Because of that , if there are no match following the lookup, we will still return error. For example, 

qaa-Cyrl-CH ;	FAIL ;	 ;	

This is what the algorithm requested (because the spec currently does not have an early return for LSR)
but in ICU and ICU4X implementation, we currently just reutrn. But why should we returan error for this case? How about if we have qaa-Cyrl-CH-u-nu-thai? we should just return qaa-Cyrl-CH-u-nu-thai right?